tools/autograd/gen_autograd.py

```diff
@@ -177,11 +104,11 @@ def is_operator_selected_for_training(decl):
     # Generate VariableType.h/cpp
     if not disable_autograd:
         from .gen_variable_type import gen_variable_type
-        gen_variable_type(out, aten_decls, differentiability_infos, template_path)
+        gen_variable_type(out, native_functions_path, differentiability_infos, template_path, operator_selector)
 
-        from . import gen_trace_type
+        from .gen_trace_type import gen_trace_type
```

Comment:
It's probably worth pulling the inline imports (here and line 106, at least) out to the top if you're back in here before landing.

Reply:
fwiw, the codegen is still simple enough and free of BC concerns that we should be able to come up with some organization that doesn't have circular problems. But I agree, it doesn't matter for codegen, inline import or whatever you want to fix the problem is fine.

Reply:
> Curious - what's the problem with inline import? I remember you mentioned we discovered some performance issue with inline import, but why it matters for codegen?

Oh definitely not a perf concern here (or most places), just hygiene.